homeassistant/components/microbees/__init__.py

```diff
@@ -0,0 +1,86 @@
+"""The microBees integration."""
+
+from dataclasses import dataclass
+from http import HTTPStatus
+import logging
+
+import aiohttp
+from microBeesPy.bee import Bee
+from microBeesPy.microbees import MicroBees
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client, config_entry_oauth2_flow
+
+from . import api
+from .const import ACCESS_TOKEN, AUTH, BEES, CONNECTOR, COORDINATOR, DOMAIN, PLATFORMS
+from .coordinator import MicroBeesUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class HomeAssistantMicroBeesData:
+    """Microbees data stored in the Home Assistant data object."""
+
+    client: MicroBees
+    bees: list[Bee]
+    session: config_entry_oauth2_flow.OAuth2Session
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, domain=DOMAIN
+) -> bool:
+    """Set up microBees from a config entry."""
+    hass.data[DOMAIN] = {}
+    hass.data.setdefault(DOMAIN, {})
+    implementation = (
+        await config_entry_oauth2_flow.async_get_config_entry_implementation(
+            hass, entry
+        )
+    )
+
+    session = config_entry_oauth2_flow.OAuth2Session(hass, entry, implementation)
+    try:
+        await session.async_ensure_token_valid()
+    except aiohttp.ClientResponseError as ex:
+        if ex.status in (
+            HTTPStatus.BAD_REQUEST,
+            HTTPStatus.UNAUTHORIZED,
+            HTTPStatus.FORBIDDEN,
+        ):
+            raise ConfigEntryAuthFailed("Token not valid, trigger renewal") from ex
+        raise ConfigEntryNotReady from ex
+
+    hass.data[DOMAIN][entry.entry_id] = {
+        AUTH: api.ConfigEntryAuth(aiohttp_client.async_get_clientsession(hass), session)
+    }
+    microbees = MicroBees(token=session.token[ACCESS_TOKEN])
+
+    hass.data[DOMAIN][CONNECTOR] = microbees
+    try:
+        bees = await microbees.getBees()
+        hass.data[DOMAIN][BEES] = bees
```

Comment:
```suggestion
```

Reply:
this is needed for setup entities

Reply:
Yes but you already store it in `hass.data[DOMAIN][entry.entry_id]` later on, so no need to do it separate here